devcycle_python_sdk/managers/sse_manager.py

```diff
@@ -0,0 +1,63 @@
+import threading
+
+import ld_eventsource
+import ld_eventsource.actions
+import ld_eventsource.config
+from typing import Callable
+
+
+class SSEManager:
+    def __init__(
+        self,
+        handlestate: Callable[[ld_eventsource.actions.Start], None],
+        handleerror: Callable[[ld_eventsource.actions.Fault], None],
+        handlemessage: Callable[[ld_eventsource.actions.Event], None],
+    ):
+        self.client: ld_eventsource.SSEClient = None
+        self.url = ""
+        self.handlestate = handlestate
+        self.handleerror = handleerror
+        self.handlemessage = handlemessage
+
+        self.read_thread = threading.Thread(
```

Comment:
How does the threading here work?

Reply:
The SSE Manager itself creates a thread managed by it - when a new url is found - it creates/updates the current thread used for processing the events incoming. 

This is mostly required due to the eventsource client must run in its own thread - and you must read from the same thread that you started the client in.